polygeist/tools/cgeist/Test/Verification/sycl/issue7835.cpp

```diff
@@ -24,7 +24,11 @@
 // CHECK-MLIR: %9 = sycl.call @getElement(%8) {MangledFunctionName = @_ZN4sycl3_V16detail7Builder10getElementILi1ELb1EEEDTcl7getItemIXT_EXT0_EEEEPNS0_4itemIXT_EXT0_EEE, TypeName = @Builder} : (memref<?x!sycl_item_1_, 4>) -> !sycl_item_1_
 // CHECK-MLIR: %memspacecast_5 = memref.memory_space_cast %cast_3 : memref<?x!llvm.struct<(!sycl_range_1_, !llvm.struct<(memref<?xi32, 4>)>)>> to memref<?x!llvm.struct<(!sycl_range_1_, !llvm.struct<(memref<?xi32, 4>)>)>, 4>
 // CHECK-MLIR: affine.store %9, %alloca[0] : memref<1x!sycl_item_1_>
-// CHECK-MLIR: sycl.call @"operator()"(%memspacecast_5, %cast) {MangledFunctionName = @_ZNK4sycl3_V16detail18RoundedRangeKernelINS0_4itemILi1ELb1EEELi1EZ4testRNS0_5queueEEUlNS0_2idILi1EEEE_EclES4_, TypeName = @RoundedRangeKernel} : (memref<?x!llvm.struct<(!sycl_range_1_, !llvm.struct<(memref<?xi32, 4>)>)>, 4>, memref<?x!sycl_item_1_>) -> ()
+// CHECK-MLIR-NEXT: %c0_6 = arith.constant 0 : index
+// CHECK-MLIR-NEXT: %10 = "polygeist.subindex"(%memspacecast_5, %c0_6) : (memref<?x!llvm.struct<(!sycl_range_1_, !llvm.struct<(memref<?xi32, 4>)>)>, 4>, index) -> memref<?x!sycl_range_1_, 4>
+// CHECK-MLIR-NEXT: %c1_7 = arith.constant 1 : index 
+// CHECK-MLIR-NEXT: %11 = "polygeist.subindex"(%memspacecast_5, %c1_7) : (memref<?x!llvm.struct<(!sycl_range_1_, !llvm.struct<(memref<?xi32, 4>)>)>, 4>, index) -> memref<?x!llvm.struct<(memref<?xi32, 4>)>, 4> 
+// CHECK-MLIR: sycl.call @"operator()"(%10, %11, %cast) {MangledFunctionName = @_ZNK4sycl3_V16detail18RoundedRangeKernelINS0_4itemILi1ELb1EEELi1EZ4testRNS0_5queueEEUlNS0_2idILi1EEEE_EclES4_, TypeName = @RoundedRangeKernel} : (memref<?x!sycl_range_1_, 4>, memref<?x!llvm.struct<(memref<?xi32, 4>)>, 4>, memref<?x!sycl_item_1_>) -> () 
```

Comment:
Do we want to avoid peeling struct type without scalar element type?

Reply:
ok, would be nice to have a list of things we plan to do in future PRs.

Reply:
We never have a formal list. Some things that comes to mind:
  - restrict qualify peeled arguments
  - handle candidate call sites even if the callee has more than one caller
  - limit the number of  struct member that are peeled (peeling too aggressively increases the number of arguments too much)
  - limit the type of members that are peeled (to perhaps just scalars)